gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/BuildTaskHelper.java

```diff
@@ -0,0 +1,75 @@
+import org.eclipse.jkube.kit.config.service.EnricherManager;
+import org.eclipse.jkube.kit.config.service.JKubeServiceHub;
+
+import static org.eclipse.jkube.kit.build.service.docker.DockerAccessFactory.DockerAccessContext.DEFAULT_MAX_CONNECTIONS;
+import static org.eclipse.jkube.kit.build.service.docker.ImagePullManager.createImagePullManager;
+
+public class BuildTaskHelper {
+  private BuildTaskHelper() { }
+
+  public static BuildServiceConfig.BuildServiceConfigBuilder buildServiceConfigBuilder(KubernetesExtension kubernetesExtension, JavaProject javaProject) {
+    return BuildServiceConfig.builder()
+      .buildRecreateMode(BuildRecreateMode.fromParameter(kubernetesExtension.getBuildRecreate().getOrElse("none")))
+      .jKubeBuildStrategy(kubernetesExtension.getBuildStrategy())
+      .forcePull(kubernetesExtension.getForcePull().getOrElse(false))
+      .buildDirectory(javaProject.getBuildDirectory().getAbsolutePath());
+  }
+
+  public static JKubeServiceHub.JKubeServiceHubBuilder initJKubeServiceHubBuilder(JKubeServiceHub.JKubeServiceHubBuilder builder, KubernetesExtension kubernetesExtension, JavaProject javaProject, KitLogger kitLogger, EnricherManager enricherManager, BuildServiceConfig.BuildServiceConfigBuilder buildServiceConfigBuilder) {
+    DockerAccess access = null;
+    if (kubernetesExtension.isDockerAccessRequired()) {
+      DockerAccessFactory.DockerAccessContext dockerAccessContext = DockerAccessFactory.DockerAccessContext.builder()
+        .log(kitLogger)
+        .projectProperties(javaProject.getProperties())
+        .maxConnections(kubernetesExtension.getMaxConnections().getOrElse(DEFAULT_MAX_CONNECTIONS))
+        .dockerHost(kubernetesExtension.getDockerHost().getOrNull())
+        .certPath(kubernetesExtension.getCertPath().getOrNull())
+        .machine(kubernetesExtension.machine)
+        .minimalApiVersion(kubernetesExtension.getMinimalApiVersion().getOrNull())
+        .skipMachine(kubernetesExtension.getSkipMachine().getOrElse(false))
+        .build();
+      DockerAccessFactory dockerAccessFactory = new DockerAccessFactory();
+      access = dockerAccessFactory.createDockerAccess(dockerAccessContext);
+    }
+    ServiceHubFactory serviceHubFactory = new ServiceHubFactory();
+    LogOutputSpecFactory logSpecFactory = new LogOutputSpecFactory(true, true, null);
+    builder.dockerServiceHub(serviceHubFactory.createServiceHub(access, kitLogger, logSpecFactory));
+    ImagePullManager imagePullManager = createImagePullManager(
+      kubernetesExtension.getImagePullPolicy().getOrElse("Always"),
+      kubernetesExtension.getAutoPull().getOrElse("true"),
+      javaProject.getProperties());
+    builder.buildServiceConfig(buildServiceConfigBuilder
+      .imagePullManager(imagePullManager)
+      .enricherTask(e -> {
+        enricherManager.enrich(PlatformMode.kubernetes, e);
+        enricherManager.enrich(PlatformMode.openshift, e);
+      })
```

Comment:
Let's take a look at this together, but since this builder is generated downstream it might make more sense to configure it there.